src/main/scala/sparkz/core/network/NetworkController.scala

```diff
@@ -142,7 +142,7 @@ class NetworkController(settings: NetworkSettings,
   }
 
   private def connectionEvents: Receive = {
-    case Connected(remoteAddress, localAddress) if connectionForPeerAddress(remoteAddress).isEmpty =>
+    case Connected(remoteAddress, localAddress) if isNewConnectionAndStillHaveRoom(remoteAddress) =>
```

Comment:
below we have a case
```
case Connected(remoteAddress, _) =>
      log.warn(s"Connection to peer $remoteAddress is already established")
      sender() ! Close
```
we need to change message of `log.warn` because now it can be one of two:
1) connection is already established
2) max connections is reached

So, let's refactor the code a bit to clearly display the cause of disconnection.